### DISCUSSION TITLE
Computed key

### DISCUSSION THREAD

**Discussion by Jack-Works** [Jack-Works/proposal-enum#11] [General]
Should the key be computed property key?

```ts
enum Computed {
    [expr]: 1,
    ["key that include space"]: 2
}
```

## Problem

The full dynamic computed property key conflicts with inline-able (#4) and hoistable declaration (#18).

It also conflicts with `[Symbol.iterator]` (#17).

Space or special character inside an enum is not common. Excluding this case can simplify the proposal.

Should we support this case?

**Comment by bathos** [comment]
> Space or special character inside an enum is not common. Excluding this case can simplify the proposal.

Supporting spaces and other characters outside the range of IdentifierName seems unrelated to the question of whether computed property keys are supported.

```js
void {
  notComputed: 1,
  "also not computed": 2,
  3: "neither was this one",
  ["this one is computed"]: 4
}
```

The first three there are all alternatives of [LiteralPropertyName](https://tc39.es/ecma262/#prod-LiteralPropertyName) (IdentifierName, StringLiteral, NumericLiteral). The last is [ComputedPropertyName](https://tc39.es/ecma262/#prod-ComputedPropertyName), a bracketed AssignmentExpression. Only the last has observable runtime semantics.

(One could probably argue that the quoted form is “least computed” in a sense — or “least sugary” is probably a better way to put it. Property keys are strings and symbols, not identifiers and numbers and strings and symbols.)
